### PR TITLE
group flag

### PR DESCRIPTION
**Pull request by bkrmendy**
## Description
This PR adds a new comment flag and a special property, so that we can designate elements as "groups". This way, we can decouple the "implementation" of a group from group behaviours (i.e., the specific code construct implementing the group behaviours in a certain context from the group behaviours themselves). Additionally, once an element is marked as a group, the editor can apply the group behaviours proactively.

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts [hunk 1]
```diff
@@ -162,3 +167,33 @@ export function treatElementAsContentAffecting(
 ): boolean {
   return getElementContentAffectingType(metadata, allElementProps, path) != null
 }
+
+export const GroupFlagKey = 'data-group'
+
+export function isGroupPragmaApplied(
```
Comment (balazsbajorics) [inline editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts:173]: I think pragma has a ton of baggage, I'd pick a name which is a little more on the safe side about using big words  

```suggestion
export function isElementMarkedAsGroup(
```

Comment (bkrmendy) [reply editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts]: makes perfect sense, renamed in https://github.com/concrete-utopia/utopia/pull/3504/commits/c79e4f708be85ebe7f46c7c022cd725b33eb186e

File: editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts [hunk 2]
```diff
@@ -162,3 +167,33 @@ export function treatElementAsContentAffecting(
...
+
+  if (isJSXElement(instance.element.value)) {
+    return (
+      instance.element.value.props.find(
+        (prop) =>
+          prop.type === 'JSX_ATTRIBUTES_ENTRY' &&
+          prop.key === GroupFlagKey &&
+          prop.value.type === 'ATTRIBUTE_VALUE' &&
+          prop.value.value === true,
+      ) != null
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts:194]: I'm sure we have a function for checking the value of a prop

Comment (balazsbajorics) [reply editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts:194]: oh crap I totally missed this! yes we have very good helpers for this :)

Comment (bkrmendy) [reply editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts:194]: I wish we had chatgpt or something to pick this up automatically 😅 any pointers about what this function might be?

Comment (Rheeseyb) [reply editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts:194]: Ok I think the best function to use here is [`getSimpleAttributeAtPath`](https://github.com/concrete-utopia/utopia/blob/master/editor/src/core/model/element-metadata-utils.ts#L2304-L2323>)

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts (hunk 2 above)
Comment (bkrmendy) [reply editor/src/components/canvas/canvas-strategies/strategies/group-like-helpers.ts]: thanks! added in https://github.com/concrete-utopia/utopia/pull/3504/commits/e8296ed05698f78e61c9aaece04d7eea74524218

**Review by ruggi — APPROVED** [review]

**Pull request merged**